resource_processor/vmss_porter/runner.py

```diff
@@ -82,40 +85,64 @@ async def filter_parameters_not_needed_by_porter(msg_body, env_vars):
     if stdout:
         result_stdout = stdout.decode()
         porter_explain_parameters = json.loads(result_stdout)["parameters"]
-        items = [item["name"] for item in porter_explain_parameters]
-        porter_keys = set(items).intersection(set(parameters.keys()))
-        return porter_keys
+        porter_parameter_keys = [item["name"] for item in porter_explain_parameters]
+        return porter_parameter_keys
     if stderr:
         result_stderr = stderr.decode()
         logger_adapter.info('[stderr]')
         for string in result_stderr.split('\n'):
             logger_adapter.info(str(string))
 
-    return parameters.keys()
+
+def get_installation_id(msg_body):
+    # this will be used to identify each bundle install within the porter state store.
+    return msg_body['id']
 
 
 async def build_porter_command(msg_body, env_vars):
+    porter_parameter_keys = await get_porter_parameter_keys(msg_body, env_vars)
     porter_parameters = ""
 
-    porter_keys = await filter_parameters_not_needed_by_porter(msg_body, env_vars)
-    for parameter in porter_keys:
-        porter_parameters = porter_parameters + f" --param {parameter}=\"{msg_body['parameters'][parameter]}\""
+    for parameter_name in porter_parameter_keys:
+        # first try to find the param in the msg body parameters collection
+        parameter_value = msg_body['parameters'].get(parameter_name)
```

Comment:
parameter_value = msg_body['parameters'].get(parameter_name, env_vars.get(parameter_name, msg_body.get(parameter_name))) I think would do the same without all the checking for None.